app/frontend/src/features/auth/useAuthStore.ts

```diff
@@ -60,12 +60,16 @@ export default function useAuthStore() {
         setLoading(true);
         try {
           const auth = await service.user.passwordLogin(username, password);
-          setJailed(auth.jailed);
 
           if (!auth.jailed) {
             const user = await service.user.getUser(username);
-            setUser(user);
+            setUserId(user.userId);
+            queryClient.setQueryData(["user", user.userId], user);
           }
+          //this must come after setting the userId, because calling setQueryData
+          //will also cause that query to be background fetched, and it needs
+          //userId to be set.
+          setJailed(auth.jailed);
```

Comment:
Not sure I 100% follow so mostly curious - what happened when you had this line before `setUserId`?

Reply:
It was fetching a user query with id = null and causing problems